src/test/java/org/jenkinsci/plugins/workflow/testMetaStep/StateMetaStep.java

```diff
@@ -0,0 +1,71 @@
+package org.jenkinsci.plugins.workflow.testMetaStep;
+
+import com.google.inject.Inject;
+import hudson.Extension;
+import hudson.model.TaskListener;
+import org.jenkinsci.Symbol;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution;
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+/**
+ * Meta-step that executes {@link State}
+ *
+ * @author Kohsuke Kawaguchi
+ */
+public class StateMetaStep extends AbstractStepImpl {
+
+    public final State state;
+    public boolean moderate;
+
+    @DataBoundConstructor
+    public StateMetaStep(State state) {
+        this.state = state;
+    }
+
+    @DataBoundSetter
+    public void setModerate(boolean m) {
```

Comment:
Might be useful to also add an (unrelated) `moderate` parameter to one of the `State` implementations, so as to verify that

```groovy
california ocean: 'pacific', mountain: 'sierra', moderate: true
```

behaves the same as

```groovy
state state: california(ocean: 'pacific', mountain: 'sierra'), moderate: true
```

but differently from

```groovy
state state: california(ocean: 'pacific', mountain: 'sierra', moderate: true)
```